doc/development/tutorials/examples/todo.py

```diff
@@ -15,106 +25,115 @@ class todolist(nodes.General, nodes.Element):
     pass
 
 
-def visit_todo_node(self, node):
+@dataclass
```

Comment:
I'd suggest using a NamedTuple here.

Reply:
could do. Or a typeddict.

Why woud you prefer a tuple here?

Reply:
Dataclass creation is costly and has mutable fields by default (which we don't want in this case I think). A TypedDict is fine as well.

Reply:
> Dataclass creation is costly and has mutable fields by default (which we don't want in this case I think). A TypedDict is fine as well (the reason why I would have used a namedtuple is just because you can access typed attributes using '.' in contrast to dict keys).

you can also mark dataclasses as 'frozen' which should be equivalent.

I guess in this simple example a named tuple will do the trick. Perhaps in more complex examples the other methods you get for free in a dataclass would come in handy. Will update to a named tuple

Reply:
> you can also mark dataclasses as 'frozen' which should be equivalent.

You also need `slots=True` but in the end, it's just easier to have a NamedTuple I think.